order/repository.go

```diff
@@ -79,6 +80,16 @@ func Find(query *bson.M, customProvider OrderCustomProvider) (iter func() (o *Or
 }
 
 func UpsertOrder(o *Order) error {
+	errUpsert := _cachedUpsertOrder(o)
+	if errUpsert != nil {
+		return errUpsert
+	}
+
+	// @todo can we update the cache entry directly with given order? (check version increment, ...)
+	return RemoveOrderCacheEntry(o.Id)
+}
+
+func _cachedUpsertOrder(o *Order) error {
```

Comment:
Uhm, does it really have to have a _ in front of the function name?

Reply:
I'd suggest to remove the underscore as well.

Let it be a private function and mention any specifics in a comment above the function definition.

Reply:
Well, lowercase is something coming from python to declare a private method since it doesn't have a good mechanism for that. For private "go" methods, lowercase letter is enough.